go/src/koding/kites/kloud/utils/object/printer.go

```diff
@@ -0,0 +1,113 @@
+package object
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/fatih/structs"
+)
+
+type Printer struct {
+	Tag    string
+	JSON   bool
+	Format string
+	W      io.Writer
+}
+
+func (p *Printer) Print(v interface{}) error {
+	if p.JSON {
+		return p.printJSON(v)
+	}
+	return p.printTab(v)
+}
+
+func (p *Printer) printJSON(v interface{}) error {
+	enc := json.NewEncoder(p.W)
+	enc.SetIndent("", "\t")
+
+	return enc.Encode(v)
+}
+
+var (
+	fprintf  = reflect.ValueOf(fmt.Fprintf)
+	stringer = reflect.TypeOf((*fmt.Stringer)(nil)).Elem()
+)
+
+func (p *Printer) printTab(v interface{}) error {
```

Comment:
@cihangir Agreed, JSONPath and text/template support (`kubectl -t '{{.Something}}'`) are really cool features for scripting with tool. For now they can be achieved with `jq` since our backend API does not support patching / server-side filtering - so there's no difference whether we do:

```
kd stack list | jq .Name
```

or

```
kd stack list -jsonpath .Name
````